rsocket-transport-aeron/src/main/java/io/rsocket/transport/aeron/AeronClientTransport.java

```diff
@@ -0,0 +1,231 @@
+                      sink -> {
+                        FragmentHandler fragmentHandler =
+                            (buffer, offset, length, header) -> {
+                              if (PayloadCodec.frameType(buffer, offset) != FrameType.SETUP_COMPLETE
+                                  || SetupCodec.connectionId(buffer, offset) != connectionId) {
+                                sink.error(new IllegalStateException("Received unexpected frame"));
+                                return;
+                              }
+
+                              final int serverStreamId = SetupCodec.streamId(buffer, offset);
+                              final String serverChannel = SetupCodec.channel(buffer, offset);
+
+                              final ExclusivePublication publication =
+                                  aeron.addExclusivePublication(
+                                      ChannelUri.addSessionId(serverChannel, (int) connectionId),
+                                      serverStreamId);
+
+                              final Subscription subscription =
+                                  aeron.addSubscription(
+                                      ChannelUri.addSessionId(channel, (int) connectionId),
+                                      streamId,
+                                      new AvailableImageHandler() {
+                                        boolean createdConnection = false;
+
+                                        @Override
+                                        public void onAvailableImage(Image image) {
+                                          if (!createdConnection) {
+                                            createdConnection = true;
+                                            sink.success(
+                                                new AeronDuplexConnection(
+                                                    serverStreamId,
+                                                    publication,
+                                                    image.subscription(),
+                                                    eventLoopGroup,
+                                                    allocator,
+                                                    prefetch,
+                                                    effort));
+                                          }
+                                        }
+                                      },
+                                      image -> {});
+                              sink.onCancel(
+                                  () -> CloseHelper.quietCloseAll(subscription, publication));
+                            };
+
+                        final NanoClock nanoClock = aeron.context().nanoClock();
+                        final long nowNs = nanoClock.nanoTime();
+                        final long deadlineNs =
+                            nowNs + aeron.context().keepAliveIntervalNs() * 1000;
```

Comment:
why * 1000 here.  Is everything in ns anyway?

Reply:
Yeah, left from experiments. Should be fixed

Reply:
fixed